### PR TITLE
Prevent default space press on image resize

### PR DESCRIPTION
**Pull request by rvantonisse-accessibility**
The spacebar default action within web document context, is to scroll the page. Therefore it is required to prevent this when triggering another action with the spacebar, within the context of the question image's resize (scale) action.

You also might need to consider to listen for key modifiers, like shift, ctrl, alt and meta, since these may trigger other actions as well. The Shift modifier for example, is used for reverse actions. For spacebar press with shift that would mean to scroll up instead of scrolling down.

### CODE REVIEW

**Comment by fnoks** [comment]
Thanks for your PR :+1: The preventDefault() is normally something we add when we do actions on "spacebar press". But for this case, it slipped through, unfortunately.

I don't think we use the key modifiers anywhere to ignore the "click", but it makes sense to me that we should. On the other hand, even the examples from w3.org ignores the key modifiers. Do you know if this is mentioned somewhere in WCAG or similar?

**Comment by icc** [comment]
Hi @rvantonisse-accessibility 
Just to verify this, in which browser did you get the scroll down when using space? Typically, by default, space should activate the focused button and not scroll the page down. I imagine this could vary between browsers though.

**Comment by rvantonisse-accessibility** [comment]
Great! 

> I don't think we use the key modifiers anywhere to ignore the "click", but it makes sense to me that we should. On the other hand, even the examples from w3.org ignores the key modifiers. Do you know if this is mentioned somewhere in WCAG or similar?

One criterion that comes into my mind is [2.1.4 Character Key Shortcuts](https://www.w3.org/TR/WCAG21/#character-key-shortcuts). But that is for single character key presses and a space is not one of them and if it was it would still pass. Also I think this is not mentioned in the WCAG specificly. You should not need to add modifier keys to prevent default behavior.

Another criterion that comes close is [3.2.2 On Input](https://www.w3.org/TR/WCAG21/#on-input), due to the unexpected behavior of the button. But scrolling does not fall under a change of context, I think... 

There is however this [first principle of ARIA's best practices](https://www.w3.org/TR/wai-aria-practices-1.1/#no_aria_better_bad_aria) that states you should cover every functionality and styling whenever you say some div is a button (or any other role) that belongs to a button.

tldr; This is just an ARIA best practice. It may be safer to use  a native html `<button type="button">` in this case.

**Comment by rvantonisse** [comment]
Hey 👋 ,

ℹ️ For your information:
checking in here as the "Ghost 👻 " commenter. I have merged my work related account, rvantonisse-accessibility, with my personal one.

Cheers,
Roel
